client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -53,9 +54,11 @@
 
     private final StreamDataRetention retentionType;
 
-    private final long generation;
+    @Setter
+    private long generation;
 
-    private final UUID readerGroupId;
+    @Setter
+    private UUID readerGroupId;
```

Comment:
This is a public facing object and is immutable. I see no reason to change that. 
https://github.com/pravega/pravega/wiki/Contributing#threading

Reply:
Removed the Setters